blockdevice/partition/gpt/gpt.go

```diff
@@ -295,6 +295,78 @@ func (g *GPT) InsertAt(idx int, size uint64, setters ...PartitionOption) (*Parti
 	return partition, nil
 }
 
+// InsertAtOffset inserts partition starts from offset bytes.
+//
+// If idx == 0, it inserts new partition as the first partition, etc., idx == 1 as the second, etc.
+func (g *GPT) InsertAtOffset(idx int, offset, size uint64, setters ...PartitionOption) (*Partition, error) {
```

Comment:
this seems to be really close to the `InsertAt` above ? should they be merged? do we need both? what is the difference?

Reply:
Yes, it can be merge, thank you. 
I'll try it.

Reply:
I don't quite understand what it does though? does it enforce partition first LBA?

Reply:
Yep, I've added WithOffset options, which allows you to create partition starts from custom LBA offset (bytes offset -> LBA auto convert).

It helps to make resize unit-test, and for this https://github.com/talos-systems/talos/issues/4041